### DISCUSSION TITLE
Can I set the tempo using the toolkit "edit" function?

### DISCUSSION THREAD

**Discussion by andre2007** [rism-digital/verovio#3847] [Q&A] [answered]
Hi,

I try to find a way to set the the tempo (BPM) using the toolkit functions.

Best guess so far is, I can somehow use the edit function:
https://book.verovio.org/toolkit-reference/toolkit-methods.html#edit

My best guess is, I can use it like this:

``` d
toolkit.edit(`{"midi.bpm": 60}`);
```
( I tried different attribute names, with and without @ ... , also with attribute name data.MIDIBPM...)

But the effect is, that my program just silently terminate on this line.

From the documentation I am not able to find out, what I can exactly do with edit function
and what I can't.

Is it possible to somehow specify the tempo to be used?

Kind regards
André

**Comment by lpugin** [comment, marked as answer]
The `edit` function is experimental and broken in many aspect, so do not rely on it. I added a [comment](https://github.com/rism-digital/verovio/commit/e490b5e38ed313ad3330a149ce15be993e1b7755) in the code, which will show up in the reference book in the next release.

You should edit your MEI data instead.

**Comment by andre2007** [comment]
Thanks, that is working fine.

```  xml
  <score xml:id="s1ge60ae">
               <scoreDef xml:id="s181g2vw" midi.bpm="65">
```
